lib/unleash/feature_toggle.rb

```diff
@@ -75,9 +76,48 @@ def am_enabled?(context)
       evaluate(context).enabled?
     end
 
+    def parent_dependencies_satisfied?(context)
+      return true if dependencies.empty?
+
+      dependencies.all? do |parent|
+        evaluate_parenthood(parent, context)
+      end
+    end
+
+    def evaluate_parenthood(parent, context)
+      parent_toggle = get_parent(parent["feature"])
+
+      return false if parent_toggle.nil?
+
+      return false unless parent_toggle.dependencies.empty?
+
+      evaluation_result = parent_toggle.is_enabled?(context)
+      return !evaluation_result if parent["enabled"] == false
```

Comment:
```suggestion
      return !parent_toggle.is_enabled?(context) unless parent["enabled"]
```

Reply:
evaluation_result is used throughout the function several times and parent.enabled cant be used like this in the code.